### DISCUSSION TITLE
Set maximum decimals instead of exact decimals

### DISCUSSION THREAD

**Discussion by chylex** [grafana/grafana#55603] [Dashboards, Panels & Visualizations]
Hi, I'd like to request a feature to set a maximum number of decimal places.

For example, here is a CPU usage graph:

![image](https://user-images.githubusercontent.com/3685160/191673254-d31d12ab-64b8-4dec-82d0-da9fba6a9a53.png)

I don't need to know numbers down to 5 decimal places. But if I set the "Decimals" property to 2, then every number has 2 decimal places even though it doesn't need them (`.00`), which makes it difficult to read, and clutters the left side of the graph which doesn't need decimal places anywhere:

![image](https://user-images.githubusercontent.com/3685160/191673563-b523dd66-78de-4152-a5fe-864e4dc3bb48.png)

I would like an option to treat the "Decimals" property as a maximum, so instead of `X.00` it would show just `X`, and instead of `X.Y0` it would show `X.Y`.

Not sure about the best UI for it, my first thought is that the "Decimals" field could accept something like `<= 2` or `..2`, but that's probably not very intuitive.

I found a similar request https://github.com/grafana/grafana/issues/8621, but that is about significant figures which would affect the left side of the decimal point too, which is something I don't want. I think both features could work simultaneously though, so you could set both a number of significant figures and then a maximum number of decimal places.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
